.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,55 @@
+name: CodeQL Analysis
+
+on:
+  push:
+    # ignore dependabot branches on push -> https://github.com/microsoft/binskim/issues/425#issuecomment-893373709
+    branches-ignore:
+      - 'dependabot/**'
+  pull_request:
+  schedule:
+    - cron: '0 8 * * *'
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: windows-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          queries: security-and-quality
+          languages: csharp
+
+      - name: Build solution
+        shell: pwsh
+        run: |
+          $pathToSolution = "ApiVersioning.sln"
+          $buildConfiguration = "Release"
+          $useSharedCompilation = "false"
+          $testProjects = @( Get-ChildItem -Path test/ -Recurse | Where-Object { $_.Extension -match 'shproj' -or $_.Extension -match 'csproj' } | ForEach-Object { $_.FullName } )
+
+          dotnet nuget locals all --clear
```

Comment:
What is the value of clearing the NuGet cache? This doesn't appear to do anything useful, except slow down the build. This _might_ be useful if local, self-referencing packages were used from build output, but this project doesn't do that. Packages are immutable with idempotent behavior.

Slowing down the build isn't the _worse_ thing in the world; however, if someone runs this locally, this is pretty significant side effect that a user/dev may not realize will happen. I'm always leery of anything that makes changes to someone's local system.